### ISSUE TITLE
Transparency color maps not correctly applied to infrared channels from certain cases

### ISSUE THREAD

**Issue by djhoese** [ssec/sift#117]
In GitLab by @jgerth on Nov 2, 2015, 15:01

The transparency color maps do not work for infrared bands from those cases that have the data subsets from the full disk (such as the 2015-10-07 case). Instead, the areas that should be transparent (or partially) are black.

**Comment by djhoese** [comment]
In GitLab by @djhoese on Nov 2, 2015, 21:52

@jgerth I tried this on my mac with some "trimmed" images and did not have a problem. I'll try the specific case you mention tomorrow on a Windows laptop and go from there.

**Comment by djhoese** [comment]
In GitLab by @djhoese on Nov 3, 2015, 12:09

@jgerth tried to show this to me, but it turns out he lies about everything. Closing for now until he can find a way to reproduce it.

**Comment by djhoese** [comment]
In GitLab by @djhoese on Nov 3, 2015, 13:45

I was able to reproduce this, but am having trouble finding a pattern. I'll document what I've found so far.

Case 1
------

1. Load B12 0000 and B14 0000 (B12 loads on top, B14 loads on bottom).
2. Set transparency color map on B14.
3. Move B14 on top of B12.
4. No transparency.

Case 2
------

1. Add B12 0000.
2. Add B14 0000 (B14 ends up on top)
3. Set transparency on B14
4. Transparency works

Case 3
------

1. Add B12 0000.
2. Add B14 0000 (B14 ends up on top)
3. Set transparency on B12
4. Move B12 on top of B14.
5. Transparency doesn't work.

Case 4
------

1. Add B08, B12, B14 at 0000 (in that order in the list)
2. Set transparent colormap on B14.
3. Move B14 on top of B12.
4. Transparency doesn't work.
5. Move B14 back underneath B12.
6. Switch B14 back to default IR colormap, set B12 to transparency colormap.
7. Transparency works.
8. Set B12 back to default IR colormap, set B08 to transparency colormap.
9. Transparency works.
10. Set B08 back to default IR colormap, set B12 to transparency colormap.
11. Move B12 on top of B08.
12. Transparency doesn't work.

Broken transparency seems to be related to reordering.

**Comment by djhoese** [comment]
In GitLab by @djhoese on Nov 3, 2015, 14:17

I've tried playing with the drawing order via the `order` property of the individual image layers at the same time the Z order is set and it has an affect, but not anything that's super useful yet. If I set the draw order so that "top" layers are drawn first then the transparency (layer 1 order < layer 2 order) doesn't work from the initial load. If I make it so the bottom layers are drawn first (layer 1 order > layer 2 order) then the initial stuff works, but moving them causes only the original relative order to be respected. So B12 moved on top of B08 and B14 will be properly transparent for just B12/B14 on, but when B08 is toggled it has no affect.
